agent/auto-config/tls.go

```diff
@@ -241,7 +241,12 @@ func (ac *AutoConfig) generateCSR() (csr string, key string, err error) {
 		conf.PrivateKeyType = connect.DefaultPrivateKeyType
 	}
 	if conf.PrivateKeyBits == 0 {
-		conf.PrivateKeyBits = connect.DefaultPrivateKeyBits
+		// If using an RSA key, a key size of at least 2048 bits is recommended; 4096 bits is better.
+		if conf.PrivateKeyType == connect.PrivateKeyTypeRSA {
+			conf.PrivateKeyBits = connect.DefaultPrivateKeyBitsRSA
+		} else {
+			conf.PrivateKeyBits = connect.DefaultPrivateKeyBits
+		}
```

Comment:
~ One other option here might be to allow passing `0` through to `GeneratePrivateKeyWithConfig` and let it negotiate the default based on the type. That might avoid repeating this logic other places or missing a 3rd key type -> bits default in the future. Happy to wait on that if you think it's best to keep the defaulting here and explicit for now.

Reply:
I like the idea of using the defaults, but I also like being specific, too. Happy to do either.

Reply:
No strong preference, glad to go w/ what you have here - more thinking out loud and noticing we haven't really centralized that type-defaulting logic. But the validation being centralized (which you have done) is the important bit.